src/components/editor/EditorDialog.tsx

```diff
@@ -0,0 +1,75 @@
+import { z } from 'zod';
+
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@/components/ui/dialog';
+
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+import { Button } from '@/components/ui/button';
+import { useDialog } from '@/lib/useDialog';
+
+interface DialogComponentProps {
+  dialog: (url: string) => void;
+}
+
+const schema = z.string().refine(val => val.startsWith('https://'), { message: 'URL must start with https' });

```

Comment:
```suggestion
const schema = z.string().url({ message: '請填入正確的網址格式, ex: https://www.journeybites.com' })).or(z.literal(''))
```
zod 有提供 url 驗證，加 `.or(z.literal(""))` 表示 optional，算是 zod 小問題，message 建議用中文(可自訂)保持跟目前風格一致

Reply:
雖然是驗心酸的，還是限制一下使用 https